addons/mail/static/src/core/common/composer.js

```diff
@@ -533,10 +533,10 @@ export class Composer extends Component {
                 });
                 for (const index in partners) {
                     const partnerData = partners[index];
-                    const persona = this.store.Persona.insert({ ...partnerData, type: "partner" });
+                    const partner = this.store["res.partner"].insert(partnerData);
                     const email = recipientEmails[index];
                     const recipient = allRecipients.find((recipient) => recipient.email === email);
-                    Object.assign(recipient, { persona });
+                    recipient.partner_id = partner.id;
```

Comment:
keep records (and not id) for as long as possible

Reply:
At this point, partner_id needs to be a Number, this is because `suggestedRecipients` and  `additionalRecipients` are not records.